File: dsa-problems/Queues/solutions/python/QUE-016-assembly-line-buffer-swap.py
from typing import List
import sys

def swap_queues(q1: List[int], q2: List[int]) -> List[List[int]]:
    return [q2, q1]

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return

    iterator = iter(input_data)
    try:
        n = int(next(iterator))
        remaining = list(iterator)

        # If we have exactly 2n values, split them in half
        if len(remaining) == 2 * n:
            q1 = [int(x) for x in remaining[:n]]
            q2 = [int(x) for x in remaining[n:]]
        # If we have exactly n values, use as q1, create q2
        elif len(remaining) == n:
            q1 = [int(x) for x in remaining]
            q2 = [0] * n  # Default second queue
        # Otherwise try to split as much as possible
        else:
            q1 = [int(x) for x in remaining[:n]]
            q2 = [int(x) for x in remaining[n:]] if len(remaining) > n else [0] * n

        result = swap_queues(q1, q2)
        for resArr in result:
            print(" ".join(map(str, resArr)))
    except (StopIteration, ValueError, IndexError):
        pass

if __name__ == "__main__":
    main()

File: dsa-problems/Queues/solutions/javascript/QUE-016-assembly-line-buffer-swap.js
const readline = require("readline");

class Solution {
  swapQueues(q1, q2) {
    return [q2, q1];
  }
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let data = [];
rl.on("line", (line) => data.push(...line.trim().split(/\s+/).filter(x => x !== "")));
rl.on("close", () => {
  if (data.length === 0) return;
  let idx = 0;
  const n = parseInt(data[idx++], 10);
  const remaining = data.slice(idx);

  let q1, q2;

  // If we have exactly 2n values
  if (remaining.length === 2 * n) {
    q1 = remaining.slice(0, n).map(x => parseInt(x, 10));
    q2 = remaining.slice(n, 2 * n).map(x => parseInt(x, 10));
  } else if (remaining.length === n) {
    // Only n values - use as q1, create default q2
    q1 = remaining.map(x => parseInt(x, 10));
    q2 = Array(n).fill(0);
  } else {
    // Fallback
    q1 = remaining.slice(0, n).map(x => parseInt(x, 10));
    q2 = remaining.length > n ? remaining.slice(n).map(x => parseInt(x, 10)) : Array(n).fill(0);
    // Pad q2 if needed
    while (q2.length < n) {
      q2.push(0);
    }
  }

  const solution = new Solution();
  const result = solution.swapQueues(q1, q2);
  result.forEach((resArr) => {
    console.log(resArr.join(" "));
  });
});
